pallets/lbp/src/lib.rs

```diff
@@ -800,11 +809,15 @@ impl<T: Config> Pallet<T> {
 		);
 
 		// zero weight at the beginning or at the end of a sale may cause a problem in the price calculation
+		// Minimum allowed weight is 2%. The exponentiation used in the math can overflow when the ration between the weights is higher than 98/2.
 		ensure!(
 			!pool_data.initial_weight.is_zero()
 				&& pool_data.initial_weight < MAX_WEIGHT
+				&& pool_data.initial_weight >= MAX_WEIGHT / 50	// 2%
 				&& !pool_data.final_weight.is_zero()
-				&& pool_data.final_weight < MAX_WEIGHT,
+				&& pool_data.final_weight < MAX_WEIGHT
+				// when initial and final weights are >= 2%, then the weights are also <= 98%
+				&& pool_data.final_weight >= MAX_WEIGHT / 50, // 2%,
 			// TODO people could leak value out the pool if initial weight is < final weight due to fee structure
```

Comment:
what this TODO mean ?

Reply:
Hmm. I don't know. I didn't add this comment.